app/src/main/java/eu/kanade/tachiyomi/ui/browse/BrowseController.kt

```diff
@@ -141,16 +150,30 @@ class BrowseController :
 
         override fun configureRouter(router: Router, position: Int) {
             if (!router.hasRootController()) {
-                val controller: Controller = when (position) {
-                    // SY -->
-                    SOURCES_CONTROLLER -> if (preferences.latestTabInFront().get()) LatestController() else SourceController()
-                    LATEST_CONTROLLER -> if (!preferences.latestTabInFront().get()) LatestController() else SourceController()
-                    // SY <--
-                    EXTENSIONS_CONTROLLER -> ExtensionController()
-                    MIGRATION_CONTROLLER -> MigrationSourcesController()
-                    else -> error("Wrong position $position")
+                if (!preferences.latestTabVisibility().get()) {
```

Comment:
This could probably moved under the assignment of the controller to deduplicate code

ie
`
val controller: Controller = if (preference) {
} else {
}
`